src/applications/proxy-rewrite/redirects/index.js

```diff
@@ -2,23 +2,56 @@ import environment from 'platform/utilities/environment';
 // Currently we only have approval for disability related ones:
 import crossDomainRedirects from './crossDomainRedirects.json';
 
+const deriveIsHostMatch = (redirect, currentWindow) => {
+  const formattedRedirectHost = redirect?.domain?.replace('www.', '')?.toLowerCase();
+  const formattedCurrentWindowHost = currentWindow?.location?.host?.replace('www.', '')?.toLowerCase();
+  return formattedRedirectHost === formattedCurrentWindowHost;
+}
+
 /*
  * Redirect to a www.va.gov page if we're on a page that's being
  * replaced
  */
-export default function redirectIfNecessary(currentWindow) {
-  const matchedRedirect = crossDomainRedirects.find(
-    redirect =>
-      redirect.domain.replace('www.', '').toLowerCase() ===
-        currentWindow.location.host.replace('www.', '').toLowerCase() &&
-      redirect.src.toLowerCase() ===
-        currentWindow.location.pathname.toLowerCase(),
+const redirectIfNecessary = (currentWindow) => {
+  // Check if the route matches an absolute cross-domain redirect.
+  const absoluteCrossDomainRedirects = crossDomainRedirects?.filter(
+    redirect => !redirect?.isCatchAll,
+  );
+  const absoluteRedirectMatch = absoluteCrossDomainRedirects?.find(
+    redirect => {
+      const isHostMatch = deriveIsHostMatch(redirect, currentWindow);
+      return isHostMatch && redirect.src.toLowerCase() === currentWindow.location.pathname.toLowerCase();
+    }
   );
 
-  if (matchedRedirect) {
+  // Redirect if it's an exact match first.
+  if (absoluteRedirectMatch) {
+    currentWindow.location.href = `${environment.BASE_URL}${
+      absoluteRedirectMatch.dest
+    }`;
+    return;
+  }
+
+  // Check if the route matches a catch-all cross-domain redirect.
+  const catchAllCrossDomainRedirects = crossDomainRedirects?.filter(
+    redirect => redirect?.isCatchAll,
+  );
+  const catchAllRedirectMatch = catchAllCrossDomainRedirects?.find(
+    redirect => {
+      const isHostMatch = deriveIsHostMatch(redirect, currentWindow);
+      const currentPathStartsWithCatchAll = currentWindow.location.pathname?.startsWith(redirect.src.toLowerCase());
+      return isHostMatch && currentPathStartsWithCatchAll;
+    };
+  );
+
+  // Redirect if if there is a catch-all match.
+  if (catchAllRedirectMatch) {
     // eslint-disable-next-line no-param-reassign
     currentWindow.location.href = `${environment.BASE_URL}${
-      matchedRedirect.dest
+      catchAllRedirectMatch.dest
     }`;
+    return;
   }
 }
+
+export default redirectIfNecessary;
```

Comment:
The entire file has great variable names. Nice work!

Reply:
Thanks so much! 🙂